src/main/java/net/snowflake/client/config/SFConnectionConfigParser.java

```diff
@@ -127,10 +124,22 @@ public static ConnectionParameters buildConnectionParameters() throws SnowflakeS
     }
   }
 
-  private static String createUrl(String account, Map<String, String> fileConnectionConfiguration) {
-    String host = String.format("%s.snowflakecomputing.com", account);
+  private static String createUrl(Map<String, String> fileConnectionConfiguration) {
+    Optional<String> maybeAccount = Optional.ofNullable(fileConnectionConfiguration.get("account"));
+    Optional<String> maybeHost = Optional.ofNullable(fileConnectionConfiguration.get("host"));
+    String host =
+        maybeHost
+            .orElse(
+                maybeAccount
+                    .map(acnt -> String.format("%s.snowflakecomputing.com", acnt))
+                    .orElse(null));
+    if(host == null || host.isEmpty()) {
+      logger.warn("Neither host nor account is specified in connection parameters");
+      return null;
```

Comment:
Is it correct situation? If yes - we shouldn't log on warn. If no - maybe we should throw exception, because it may end up in NPE?

Reply:
Actually it is incorrect. Exception added.